src/hotspot/share/gc/shared/threadLocalAllocBuffer.hpp

```diff
@@ -183,10 +183,10 @@ class ThreadLocalAllocBuffer: public CHeapObj<mtThread> {
   }
 
   // Code generation support
-  static ByteSize start_offset()                 { return byte_offset_of(ThreadLocalAllocBuffer, _start); }
-  static ByteSize end_offset()                   { return byte_offset_of(ThreadLocalAllocBuffer, _end); }
-  static ByteSize top_offset()                   { return byte_offset_of(ThreadLocalAllocBuffer, _top); }
-  static ByteSize pf_top_offset()                { return byte_offset_of(ThreadLocalAllocBuffer, _pf_top); }
+  static constexpr ByteSize start_offset()       { return byte_offset_of(ThreadLocalAllocBuffer, _start); }
+  static constexpr ByteSize end_offset()         { return byte_offset_of(ThreadLocalAllocBuffer, _end); }
+  static constexpr ByteSize top_offset()         { return byte_offset_of(ThreadLocalAllocBuffer, _top); }
+  static constexpr ByteSize pf_top_offset()      { return byte_offset_of(ThreadLocalAllocBuffer, _pf_top); }
```

Comment:
Is constexpr really necessary here?  Isn't this usually a constant expression for most compilers?

Reply:
We have many of these inline functions that could be either `const` and `constexpr`. Since `constexpr` has been allowed only recently, there are very few functions that are actually declared `constexpr`.

My recommendation is to limit `constexpr` only to new code. We can change `const` to `constexpr` when required (e.g., when an existing function is called by new code that's in a `constexpr` context). That way, we can limit code churn.

Reply:
I agree and it's more characters to look at for no real performance benefit.  These offset functions are in many classes.